openstack/utils/templates/_helpers.tpl

```diff
@@ -60,3 +60,25 @@ enabled = false
     {{- $options := merge .Values.osprofiler .Values.global.osprofiler }}
     {{- if $options.enabled }} osprofiler{{ end -}}
 {{- end }}
+
+{{- define "jaeger_agent_sidecar" }}
```

Comment:
This is a helper function defining the jaeger agent side car container. The version of the Jaeger-agent image must be the same as the Jaeger operator. So I add the jaeger operator as the subchart and import the version number from there.

The agent collects the profiling data, and send them to the openstack-jaeger-collector.

Reply:
What happens if the agent doesn't have the same version number anymore? Does it crash? Will it bring down cinder-api, because one of the containers crashed?
It's fine, if you don't have answers, yet, but please figure this out.

Reply:
good point, I don't know the answer yet, will figure it out.

Reply:
what happens when the agent is down?

Reply:
To summarize the questions on the sidecar.

I have not found any document on the behavior of agent. My bet is some warning or error message spit out, either from the agent when posting the message to collector, or from the collector side when parsing the message. I need to spend more time to find out how jaeger behaves. 

But what if the agent really crashes due to some reason? In this case, a custom liveness probe should be set. For cinder api, it is set to 

> Liveness:   http-get http://:cinder-api/ delay=15s timeout=5s period=10s #success=1 #failure=3

K8s would not restart the pod when the agent crashes, but only when cinder api fails. There is already another sidecar in the cinder-api pod, the statsd exporter.